### PR TITLE
Juanro/employee termination implementation

### PR DESCRIPTION
**Pull request by Juanro-Minne**
- Added termination functionality and resigned status on the profile
- Added validation on the form and implemented design
- Added fix for bug where client allocated none was displayed twice 
image

image
image
image
image
image

-Below are the screenshots for the roles and having access to terminations
image
TALENT

image
EMPLOYEE

image
SUPER ADMIN/ADMIN

### CODE REVIEW

File: src/app/components/hris/employees/employee-termination/employee-termination.component.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      dayOfNotice: newterminationform.dayOfNotice,
+      lastDayOfEmployment: newterminationform.lastDayOfEmployment,
+      reemploymentStatus: newterminationform.reEmploymentStatus,
+      equipmentStatus: newterminationform.equipmentStatus,
+      accountsStatus: newterminationform.accountsStatus,
+      terminationDocument: this.base64String,
+      documentName: newterminationform.terminationDocument,
+      terminationComments: newterminationform.terminationComments
+    };
+    console.log(employeeTerminationDto)
```
Comment (MatthewSchoeman) [inline src/app/components/hris/employees/employee-termination/employee-termination.component.ts]:
```suggestion
    
```

File: src/app/components/hris/employees/employee-termination/employee-termination.component.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    this.newterminationform = this.fb.group({
+      terminationOption: new FormControl('', Validators.required),
+    dayOfNotice: new FormControl<Date | string>(new Date(Date.now()), Validators.required),
+    lastDayOfEmployment: new FormControl<Date | string>(new Date(Date.now()), Validators.required),
+    reEmploymentStatus: new FormControl<boolean>(false, Validators.required),
+    equipmentStatus: new FormControl<boolean>(false, Validators.required),
+    accountsStatus: new FormControl<boolean>(false, Validators.required),
+    terminationDocument: new FormControl<string>('', Validators.required),
+    terminationComments: new FormControl<string>(''),
+  });
```
Comment (MatthewSchoeman) [inline src/app/components/hris/employees/employee-termination/employee-termination.component.ts:61]: Format this to keep a consistent alignment

File: src/app/components/hris/employees/employee-profile/employee-profile.component.html [hunk 3]
```diff
@@ -383,4 +390,4 @@ <h5 class="col-2">{{documentFormProgress}}%</h5>
```
Comment (madwizzard) [inline src/app/components/hris/employees/employee-profile/employee-profile.component.html:390]:
```suggestion
```

File: src/app/components/hris/employees/employee-termination/employee-termination.component.ts (hunk 2 above)
Comment (Juanro-Minne) [reply src/app/components/hris/employees/employee-termination/employee-termination.component.ts]: done

**Comment by MatthewSchoeman** [comment]
image


Is the alignment for the radio button and checkboxes correct because they are misaligned?

**Comment by Juanro-Minne** [comment]
> image
> 
> Is the alignment for the radio button and checkboxes correct because they are misaligned?

image

**Comment by MatthewSchoeman** [comment]
image


Should off boarding tasks not be left aligned with the previous heading?

**Comment by Juanro-Minne** [comment]
> image
> 
> Should off boarding tasks not be left aligned with the previous heading?

image

**Comment by MatthewSchoeman** [comment]
image

The icon and text are misaligned and very close to each other. Please confirm with the figma design

**Comment by MatthewSchoeman** [comment]
image

Should these button be different sizes?

**Comment by Juanro-Minne** [comment]
> image
> 
> Should these button be different sizes?

image

**Comment by MatthewSchoeman** [comment]
image

The start date has different styling to the rest of the info icons. It is lower aligned and the value has a different styling

**Comment by Juanro-Minne** [comment]
> image
> 
> The start date has different styling to the rest of the info icons. It is lower aligned and the value has a different styling

image

**Comment by Juanro-Minne** [comment]
> > image
> > The start date has different styling to the rest of the info icons. It is lower aligned and the value has a different styling
> 
> image

https://github.com/RetroRabbit/RGO-Client/pull/464#issuecomment-2151796194

**Comment by Juanro-Minne** [comment]
> > image
> > The start date has different styling to the rest of the info icons. It is lower aligned and the value has a different styling
> 
> image

image

**Review by MatthewSchoeman — APPROVED** [review]

**Pull request merged**